manim/mobject/types/vectorized_mobject.py

```diff
@@ -1060,15 +1060,28 @@ def get_points_defining_boundary(self):
         # Probably returns all anchors, but this is weird regarding  the name of the method.
         return np.array(list(it.chain(*[sm.get_anchors() for sm in self.get_family()])))
 
-    def get_arc_length(self, n_sample_points=None):
-        if n_sample_points is None:
-            n_sample_points = 4 * self.get_num_curves() + 1
-        points = np.array(
-            [self.point_from_proportion(a) for a in np.linspace(0, 1, n_sample_points)]
+    def get_arc_length(self, sample_points_per_curve: Optional[float] = None) -> float:
+        """Return the approximated length of the whole curve.
+
+        Parameters
+        ----------
+        sample_points_per_curve
+            Number of sample points per curve used to approximate the length. More points result in a better approximation.
+
+        Returns
+        -------
+        float
+            The length of the curve
+        """
+        kwargs = {}
+        if sample_points_per_curve is not None:
+            kwargs["sample_points"] = sample_points_per_curve
+        return np.sum(
+            [
+                length
+                for curve, length in self.get_curve_functions_with_lengths(**kwargs)
+            ]
         )
```

Comment:
```suggestion
        return np.sum(length for _, length in self.get_curve_functions_with_lengths(sample_points=sample_points_per_curve))
```
Might be too long for black's tastes but that's the structure I think this should follow. The use of `_, length` explicates that we don't care about the first element. And then `get_curve_function_with_length` should have `sample_points` default to `None` and then something like
```py
if sample_points is None:
    sample_points = 10
```

Reply:
I don't think it is a good idea to explicitly define a default number of sample points at two positions in the code. The reason I changed it is, that before my changes `get_arc_length` used 4 points by default and `get_point_from_proportion` used 10. We risk inconsistencies by writing default values twice.

Reply:
Please do not resolve conversations  when they are not resolved.

What I meant was that you'd change `get_curve_function_with_length` to be like
```py
def get_curve_function_with_length(self, n, sample_points=None):
    if sample_points is None:
        sample_points = 10
    ...
```
and then forward on the `sample_points_per_curve` normally as then `None` will be handled appropriately

Reply:
> Please do not resolve conversations when they are not resolved.

Oh, I am sorry.

I hope that now it is like you imagined it. I feel like this is a good way to do it :)